From 589f06bc02205d6ec3c8b21c463174b3bd81ca4a Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 30 Jul 2018 14:50:24 +0800
Subject: [PATCH 57/65] do not customize window theme

The customized window theme may broke metacity.
...
|(metacity:2675): metacity-WARNING **: "(null)" found in configuration
database is not a valid value for mouse button modifier
...

Also remove install-data-hook, which caused host contamination

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>

Rebase for anaconda 34.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 37. And it replaces metacity with gnome-kiosk in
upstream. Revert it and turn back to metacity.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 38 on 20231107.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 39 on 20240304.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase for anaconda 42 on 20250328.

Signed-off-by: Kai Kang <kai.kang@windriver.com>
---
 data/window-manager/config/Makefile.am |  6 +-----
 pyanaconda/display.py                  | 25 ++++++++++---------------
 2 files changed, 11 insertions(+), 20 deletions(-)

diff --git a/data/window-manager/config/Makefile.am b/data/window-manager/config/Makefile.am
index 90e71d9161..5d7a2b80e1 100644
--- a/data/window-manager/config/Makefile.am
+++ b/data/window-manager/config/Makefile.am
@@ -26,11 +26,7 @@ dist_schema_DATA = org.gnome.desktop.wm.keybindings.gschema.override \
 # schemas they modify, so pretend that this is the case with symlinks and
 # create the compiled schema.
 install-data-hook:
-	$(MKDIR_P) $(DESTDIR)$(schemadir)
-	$(LN_S) -f /usr/share/glib-2.0/schemas/org.gnome.desktop.wm.keybindings.gschema.xml $(DESTDIR)$(schemadir)
-	$(LN_S) -f /usr/share/glib-2.0/schemas/org.gnome.desktop.wm.preferences.gschema.xml $(DESTDIR)$(schemadir)
-	$(LN_S) -f /usr/share/glib-2.0/schemas/org.gnome.desktop.enums.xml $(DESTDIR)$(schemadir)
-	glib-compile-schemas --strict $(DESTDIR)$(schemadir)
+	echo "pass"
 
 uninstall-local:
 	rm -f $(DESTDIR)$(schemadir)/*.xml
diff --git a/pyanaconda/display.py b/pyanaconda/display.py
index 7ef61a1a95..6667d2e5be 100644
--- a/pyanaconda/display.py
+++ b/pyanaconda/display.py
@@ -198,7 +198,7 @@ def do_startup_wl_actions(timeout, headless=False, headless_resolution=None):
     if 'XDG_DATA_DIRS' in os.environ:
         xdg_data_dirs = datadir + '/window-manager:' + os.environ['XDG_DATA_DIRS']
     else:
-        xdg_data_dirs = datadir + '/window-manager:/usr/share'
+        xdg_data_dirs = '/usr/share'
 
     xdg_config_dirs = datadir
     if 'XDG_CONFIG_DIRS' in os.environ:
@@ -225,23 +225,18 @@ def do_startup_wl_actions(timeout, headless=False, headless_resolution=None):
         # local display - switch to VT7 & show GUI there
         argv.extend(["--vt", "7"])
 
-    # add the generic GNOME Kiosk invocation
-    argv.extend(["gnome-kiosk", "--sm-disable",
-                 "--wayland", "--no-x11",
-                 "--wayland-display", constants.WAYLAND_SOCKET_NAME])
+    # add the metacity invocation
+    argv.extend(["metacity", "--sm-disable",
+                 "--display", ":1"])
 
-    # remote access needs gnome-kiosk to start in headless mode
-    if headless:
-        argv.extend(["--headless"])
-
-    # redirect stdout and stderr from GNOME Kiosk to journal
-    gnome_kiosk_stdout_stream = journal.stream("gnome-kiosk", priority=journal.LOG_INFO)
-    gnome_kiosk_stderr_stream = journal.stream("gnome-kiosk", priority=journal.LOG_ERR)
+    # redirect stdout and stderr from metacity to journal
+    metacity_stdout_stream = journal.stream("metacity", priority=journal.LOG_INFO)
+    metacity_stderr_stream = journal.stream("metacity", priority=journal.LOG_ERR)
 
     childproc = util.startProgram(argv, env_add={'XDG_DATA_DIRS': xdg_data_dirs},
                                   preexec_fn=wl_preexec,
-                                  stdout=gnome_kiosk_stdout_stream,
-                                  stderr=gnome_kiosk_stderr_stream,
+                                  stdout=metacity_stdout_stream,
+                                  stderr=metacity_stderr_stream,
                                   )
     WatchProcesses.watch_process(childproc, argv[0])
 
@@ -254,7 +249,7 @@ def do_startup_wl_actions(timeout, headless=False, headless_resolution=None):
 
     WatchProcesses.unwatch_process(childproc)
     childproc.terminate()
-    raise TimeoutError("Timeout trying to start gnome-kiosk")
+    raise TimeoutError("Timeout trying to start metacity")
 
 
 def set_resolution(runres):
-- 
2.7.4
